packages/block-library/src/gallery/edit.js

```diff
@@ -32,7 +33,20 @@ import {
  */
 import GalleryImage from './gallery-image';
 
-const MAX_COLUMNS = 8;
+/**
+ * Filters the minimum number of allowed columns of the core/gallery block.
+ *
+ * @param {Number} number The column count. Defaults to 2.
+ */
+const MIN_COLUMNS = applyFilters( 'blocks.gallery.minColumns', 2 );
```

Comment:
What if I want to change the min columns of a gallery, that has certain attributes or that is nested in a specific block? I feel we should pass more information at least the block client id.